sound/soc/sof/intel/hda.c

```diff
@@ -499,6 +499,44 @@ static const struct sof_intel_dsp_desc
 	return chip_info;
 }
 
+irqreturn_t hda_dsp_interrupt_handler(int irq, void *context)
+{
+	struct snd_sof_dev *sdev = context;
+
+	if (is_ipc_irq(sdev) || is_stream_irq(sdev))
+		return IRQ_WAKE_THREAD;
+
+	return IRQ_NONE;
+}
+
+irqreturn_t hda_dsp_interrupt_threaded(int irq, void *context)
+{
+	struct snd_sof_dev *sdev = context;
+
+	if (is_ipc_irq(sdev))
+		schedule_work(&sdev->ipc_work);
+	if (is_stream_irq(sdev))
+		schedule_work(&sdev->stream_work);
+
+	return IRQ_HANDLED;
+}
+
+static void sof_ipc_work(struct work_struct *work)
+{
+	struct snd_sof_dev *sdev =
+		container_of(work, struct snd_sof_dev, ipc_work);
+
+	sof_ops(sdev)->irq_thread(0, sdev);
+}
+
+static void sof_stream_work(struct work_struct *work)
+{
+	struct snd_sof_dev *sdev =
+		container_of(work, struct snd_sof_dev, stream_work);
+
+	hda_dsp_stream_threaded_handler(0, sdev);
```

Comment:
this is really odd, you are not using the first argument 0?

Reply:
No, the first argument is never used.

Reply:
@bardliao a general comment about the naming of the functions. We should only call the hard interrupt handler as somthing_thread_handler(), the thread itself should just be called irq_thread or the stream_thread